x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/TransportTermsEnumAction.java

```diff
@@ -147,7 +148,22 @@ protected void doExecute(Task task, TermsEnumRequest request, ActionListener<Ter
         if (ccsCheckCompatibility) {
             checkCCSVersionCompatibility(request);
         }
-        new AsyncBroadcastAction(task, request, listener).start();
+        ActionListener<TermsEnumResponse> loggingListener = listener.delegateResponse((l, e) -> {
```

Comment:
After trying a bit provoking 5xxs here I think we might have to change exception handling or do something else here. In https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/TransportTermsEnumAction.java#L384 we are catching any exception and returning it inside the NodeTermsEnumResponse which then gets [merged with others](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/TransportTermsEnumAction.java#L229), so I think we cannot rely on exceptions bubbling up but would need to inspect the shardFailures in TermsEnumResponse. Problem also is they don't really have a status code anymore, all terms_enum requests succeed either complete or incomplete with a 200. I think that choice was made by design so make this API (mainly used for auto_complete in Kibana) as painless to consume as possible.